chemprop/uncertainty/predictor.py

```diff
@@ -91,11 +107,75 @@ def _calc_prediction_uncertainty(
 
 @UncertaintyPredictorRegistry.register("dropout")
 class DropoutPredictor(UncertaintyPredictor):
-    def _calc_prediction_uncertainty(
-        self, dataloader: DataLoader, models: Iterable[MPNN], trainer: pl.Trainer
-    ) -> tuple[Tensor, Tensor]:
-        ...
-        return
+    """
+    A :class:`DropoutPredictor` creates a virtual ensemble of via Monte Carlo dropout with the provided models [1]_.
+    
+    References
+    -----------
+    .. [1] arXiv:1506.02142 [stat.ML]
+    model parameters. Predicts uncertainty for predictions based on the variance in predictions among
+    an ensemble's submodels.
+    """
+
+    def __init__(self, ensemble_size: int, dropout: float):
+        """
+        Parameters
+        ----------
+        ensemble_size (int): The number of samples to draw for the ensemble.
+        dropout (float): The probability of dropping out units in the dropout layers.
+        """
+        self.ensemble_size = ensemble_size
+        self.dropout = dropout
+
+    def _calc_prediction_uncertainty(self, dataloader, models, trainer) -> Tensor:
+        if len(models) != 1:
+            raise ValueError(
+                "Dropout method for uncertainty only takes exactly one model."
+            )
+        model = next(iter(models))
+        self._setup_predict_wrapper(model)
+        individual_preds = []
+
+        for _ in range(self.ensemble_size):
+            predss = trainer.predict(model, dataloader)
+            preds = torch.concat(predss, 0)
+            if isinstance(model.predictor, MulticlassClassificationFFN):
+                preds = torch.argmax(preds, dim=-1)
+            individual_preds.append(preds)
+
+        stacked_preds = torch.stack(individual_preds, dim=0).float()
+        means = torch.mean(stacked_preds, dim=0)
```

Comment:
```suggestion
        means = torch.mean(stacked_preds, dim=0).unsqueeze(-1)
```
As the shape of predictions would be ``(n, t, m)`` for regression or binary classification and ``(n, t, c, m)`` for multiclass classification, I think the dimension for `m=1` is missed here.